### PR TITLE
[LMS-482] Revamped Workflows

### PR DESCRIPTION
**Pull request by stephenfuqua**
* Extract shared steps to reusable workflows
* Add CodeQL on pull request
* Add dependency scanner on pull request
* Update 3rd party actions
* Remove the unused file tester

You can see these actions at work [in my LMS-Toolkit fork](https://github.com/stephenfuqua/LMS-Toolkit/actions)

### CODE REVIEW

File: .github/workflows/reusable-test.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        working-directory: ${{ env.SRC_DIR }}
+
+      - name: Run CI Test Task
+        run: python ./eng/build.py ci:test ${{ env.PACKAGE_NAME }}
+
+      - name: Archive code coverage results
+        uses: actions/upload-artifact@3cea5372237819ed00197afe530f5a7ea3e805c8  # v3.1.0
+        with:
+          name: code-coverage-report
+          path: ${{ env.SRC_DIR }}/htmlcov/
```
Comment (stephenfuqua) [inline .github/workflows/reusable-test.yml:49]: The steps above originated from the various *-test.yml files, but were tweaked by changing the poetry install and adding the "poetry" caching in the setup-python step. We used that same pattern successfully in the Suite3-Performance-Testing repository.

File: .github/workflows/test-sql-adapter.yml [hunk 2]
```diff
@@ -9,55 +9,19 @@ on:
...
-
-      - name: Archive code coverage results
-        uses: actions/upload-artifact@e448a9b857ee2131e752b06002bf0e093c65e571
-        with:
-          name: code-coverage-report
-          path: ${{ env.SRC_DIR }}/htmlcov/
+    name: Canvas Extractor - Test
+    uses: ./.github/workflows/reusable-test.yml
+    with:
+      project: sql-adapter
```
Comment (stephenfuqua) [inline .github/workflows/test-sql-adapter.yml:27]: The deleted files below were for the `file-tester` utility. The readme in that directory mentioned that the application was broken at the time we released LMS Toolkit 1.0. We had no plans to fix it, so I finally just removed it from the repository.

File: .github/workflows/test-google-classroom.yml [hunk 3]
```diff
@@ -9,55 +9,19 @@ on:
...
-
-      - name: Run CI Test Task
-        run: python ./eng/build.py ci:test ${{ env.PACKAGE_NAME }}
-
-      - name: Archive code coverage results
-        uses: actions/upload-artifact@e448a9b857ee2131e752b06002bf0e093c65e571
-        with:
-          name: code-coverage-report
-          path: ${{ env.SRC_DIR }}/htmlcov/
+    name: Canvas Extractor - Test
```
Comment (saa14) [inline .github/workflows/test-google-classroom.yml:24]: SHOULD: The `name` here should match the feature being tested.

File: .github/workflows/test-lms-ds-loader.yml [hunk 4]
```diff
@@ -9,55 +9,19 @@ on:
...
-
-      - name: Run CI Unit Test Task
-        run: python ./eng/build.py ci:test ${{ env.PACKAGE_NAME }}
-
-      - name: Archive code coverage results
-        uses: actions/upload-artifact@e448a9b857ee2131e752b06002bf0e093c65e571
-        with:
-          name: code-coverage-report
-          path: ${{ env.SRC_DIR }}/htmlcov/
+    name: Canvas Extractor - Test
```
Comment (saa14) [inline .github/workflows/test-lms-ds-loader.yml:24]: SHOULD: The `name` here should match the feature being tested.

File: .github/workflows/test-lms-harmonizer.yml [hunk 5]
```diff
@@ -9,57 +9,19 @@ on:
...
-
-      - name: Run CI Test Task
-        run: python ./eng/build.py ci:test ${{ env.PACKAGE_NAME }}
-
-      - name: Archive code coverage results
-        uses: actions/upload-artifact@e448a9b857ee2131e752b06002bf0e093c65e571
-        with:
-          name: code-coverage-report
-          path: ${{ env.SRC_DIR }}/htmlcov/
+    name: Canvas Extractor - Test
```
Comment (saa14) [inline .github/workflows/test-lms-harmonizer.yml:24]: SHOULD: The `name` here should match the feature being tested.

File: .github/workflows/test-schoology.yml [hunk 6]
```diff
@@ -9,55 +9,19 @@ on:
...
-
-      - name: Run CI Test Task
-        run: python ./eng/build.py ci:test ${{ env.PACKAGE_NAME }}
-
-      - name: Archive code coverage results
-        uses: actions/upload-artifact@e448a9b857ee2131e752b06002bf0e093c65e571
-        with:
-          name: code-coverage-report
-          path: ${{ env.SRC_DIR }}/htmlcov/
+    name: Canvas Extractor - Test
```
Comment (saa14) [inline .github/workflows/test-schoology.yml:24]: SHOULD: The `name` here should match the feature being tested.

File: .github/workflows/test-sql-adapter.yml [hunk 7]
```diff
@@ -9,55 +9,19 @@ on:
...
-
-      - name: Run CI Test Task
-        run: python ./eng/build.py ci:test ${{ env.PACKAGE_NAME }}
-
-      - name: Archive code coverage results
-        uses: actions/upload-artifact@e448a9b857ee2131e752b06002bf0e093c65e571
-        with:
-          name: code-coverage-report
-          path: ${{ env.SRC_DIR }}/htmlcov/
+    name: Canvas Extractor - Test
```
Comment (saa14) [inline .github/workflows/test-sql-adapter.yml:24]: SHOULD: The `name` here should match the feature being tested.

**Review by saa14 — APPROVED** [review]

**Review by pmcvtm — APPROVED** [review]
Looks good. Great copy cleanup and nice catches on the name mismatches.

**Pull request merged**